docs/migration/4_0.md

```diff
@@ -1,3 +1,29 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+
+### Feature keys for product configurators
+
+The feature keys that are used to lazily load the product configurator libraries in app.module.ts were available as `rulebased` and `productConfiguratorRulebased` from 3.1 onwards (respective `textfield` and `productConfiguratorTextfield` for the textfield template configurator).
+
+In 4.0, only the longer versions `productConfiguratorRulebased` and `productConfiguratorTextfield` are possible.
+
+Example: A configuration
+
+      featureModules: {
+        rulebased: {
+          module: () => import('@spartacus/product-configurator/rulebased').then(
+          (m) => m.RulebasedConfiguratorModule
+        ),
+        },
+      }
+
+needs to look like that in 4.0
+
+      featureModules: {
+        productConfiguratorRulebased: {
+          module: () => import('@spartacus/product-configurator/rulebased').then(
+          (m) => m.RulebasedConfiguratorModule
+        ),
+        },
+      }
```

Comment:
nitpick: for better .md formatting, you might want to wrap code snippets in 
<pre>
```typescript
  code here
```
</pre>